src/core/shape_inference/include/tile_shape_inference.hpp

```diff
@@ -14,33 +15,51 @@ void shape_infer(const Tile* op,
+    NODE_VALIDATION_CHECK(op, repeats_shape.rank().compatible(1), "Tile repeats must be of rank 1");
+
     const auto& arg_shape = input_shapes[0];
-    auto& repeats_shape = input_shapes[1];
     auto& output_shape = output_shapes[0];
-    using DimType = typename std::iterator_traits<typename T::iterator>::value_type;
-    std::vector<int64_t> axes_val;
-    NODE_VALIDATION_CHECK(op, repeats_shape.rank().compatible(1), "PartialShape of repeats must be of rank 1");
-
-    // Get repeats
-    bool axes_are_known = get_data_as_int64<T>(1, op, axes_val, constant_data);
-    const auto arg_rank = arg_shape.rank();
-    if (arg_rank.is_static() && (axes_are_known || repeats_shape[0].is_static())) {
-        // try to specify rank
-        int64_t data_rank = arg_shape.size();
-        int64_t repeats_rank = axes_are_known ? axes_val.size() : repeats_shape[0].get_length();
-        auto output_rank = std::max(data_rank, repeats_rank);
-        output_shape.resize(output_rank);
-        // if have constant axes, compute new axes
-        if (axes_are_known) {
-            auto remain_arg = output_rank - data_rank;
-            auto remain_axes = output_rank - repeats_rank;
-            for (size_t i = 0; i < static_cast<size_t>(output_rank); i++) {
-                auto data_tmp = i < static_cast<size_t>(remain_arg) ? DimType(1) : arg_shape[i - (remain_arg)];
-                auto repeat_tmp = i < static_cast<size_t>(remain_axes) ? DimType(1) : axes_val[i - remain_axes];
-                output_shape[i] = data_tmp * repeat_tmp;
-            }
-        }
+
+    // Get repeats and pre process values
+    T repeats;
+    bool has_repeats;
+    if (auto rep_data = get_input_const_data_as<T, int64_t>(op, 1, constant_data)) {
+        // set negatives repeats to 0
+        repeats.resize(rep_data->size());
+        std::transform(rep_data->begin(), rep_data->end(), repeats.begin(), [](int64_t r) -> TDim {
+            return {static_cast<typename TDim::value_type>(std::max(static_cast<int64_t>(0), r))};
+        });
+        has_repeats = true;
+    } else {
+        has_repeats = get_data_as_shape(1, op, repeats);
+    }
+
+    const auto& arg_rank = arg_shape.rank();
+    if (arg_rank.is_static() && has_repeats) {
+        const auto output_rank = std::max(arg_shape.size(), repeats.size());
+
+        std::vector<TDim> dims;
+        dims.reserve(output_rank);
+
+        // add missing repeats
+        repeats.insert(repeats.begin(), output_rank - repeats.size(), TDim{1});
+
+        // insert missing input dimensions
+        auto rep_it = std::next(repeats.begin(), output_rank - arg_shape.size());
+        dims.insert(dims.begin(), repeats.begin(), rep_it);
+
+        // calc repeated output dimensions
+        std::transform(arg_shape.begin(), arg_shape.end(), rep_it, std::back_inserter(dims), std::multiplies<TDim>());
+
+        output_shape = T(std::move(dims));
+    } else if (arg_rank.is_static() && repeats_shape[0].is_static()) {
```

Comment:
Fails on `repeats_shape[0]` when `repeats_shape` has dynamic rank.

Reply:
#15792 should fix it.